### PR TITLE
feat: extend multimodality

### LINKED ISSUE #4481: [FEATURE] extend `multi-modality` support with `DataURL` for documents like `pdf`
**Is your feature request related to a problem? Please describe.**
We recently got a request from a user about support for PDF documents within the TextField as DataURL.
Leire found that the following worked:
`'<embed src=https://arxiv.org/pdf/2307.00808.pdf type=application/pdf width=700px height=700px/></embed>'`

**Describe the solution you'd like**
I would like to see extended support for documents like PDF, and maybe pptx, docx, xlsx and others to load them into Argilla.
`from argilla.client.feedback.utils import audio_to_html, image_to_html, video_to_html`

**Describe alternatives you've considered**
N.A.

**Additional context**
image

### PR DESCRIPTION
**Pull request by sdiazlor**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #4481

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
@sdiazlor, would you also be able to look at other documents like Excel, word etc? I believe this might work similarly and might be easy to add.

**Comment by sdiazlor** [comment]
> @sdiazlor, would you also be able to look at other documents like Excel, word etc? I believe this might work similarly and might be easy to add.

@davidberenstein1957 Yes, I was gonna test it too

**Comment by sdiazlor** [comment]
@davidberenstein1957 I thought that it would be better to only update the docs together with Leire's change, to avoid more basic functions as Dani had mentioned some time ago.

**Comment by davidberenstein1957** [comment]
@sdiazlor, I would still add a specific function for the PDF to DataURL integration but don't spend time on the docx/excel ones

**Comment by davidberenstein1957** [comment]
@sdiazlor, don't forget to update the changelog either.

**Comment by davidberenstein1957** [comment]
Perfect @sdiazlor. Did you also want to add an example to the original making most of markdown posts?

**Comment by sdiazlor** [comment]
> Perfect @sdiazlor. Did you also want to add an example to the original making most of markdown posts?

Yes, I'm on it!

**Comment by davidberenstein1957** [comment]
@sdiazlor @leiyre thanks for the quick and great implementation. People are going to be happy with this one :)

File: src/argilla/client/feedback/utils/html_utils.py [hunk 1]
```diff
@@ -145,6 +146,54 @@ def image_to_html(media_source: Union[str, bytes], file_type: Optional[str] = No
...
+        file_data = pdf_path.read_bytes()
+
+    if len(file_data) > 5000000:
+        raise ValueError(
+            f"File size is {len(file_data)} bytes. It is recommended to use files smaller than 5MB, as larger files might not render properly."
+        )
+
+    pdf_base64 = base64.b64encode(file_data).decode("utf-8")
+    data_url = f"data:application/pdf;base64,{pdf_base64}"
+    html = f'<object id="pdf" data="{data_url}" type="application/pdf" width="700" height="700"><p>Unable to display PDF file.</p></object>'
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/utils/html_utils.py]: Is thhere a reason for this width and height and not use 100%? Should we allow adding this as extra args?

File: src/argilla/client/feedback/utils/html_utils.py [hunk 2]
```diff
@@ -145,6 +146,54 @@ def image_to_html(media_source: Union[str, bytes], file_type: Optional[str] = No
...
+    else:
+        pdf_path = Path(input_file)
+
+        if not pdf_path.exists() or pdf_path.stat().st_size == 0:
+            raise FileNotFoundError(f"File {input_file} does not exist or is empty.")
+        if pdf_path.suffix.lower() != ".pdf":
+            raise ValueError("Provided file is not a PDF.")
+        file_data = pdf_path.read_bytes()
+
+    if len(file_data) > 5000000:
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/utils/html_utils.py]: can we unify this logic across other methods?

File: src/argilla/client/feedback/utils/html_utils.py [hunk 3]
```diff
@@ -145,6 +146,54 @@ def image_to_html(media_source: Union[str, bytes], file_type: Optional[str] = No
...
+
+    Examples:
+        >>> from argilla.client.feedback.utils import pdf_to_html
+        >>> html = pdf_to_html("my_pdf.pdf")
+        >>> html = pdf_to_html(b"my_pdf.pdf")
+        >>> html = pdf_to_html("https://my_pdf.pdf")
+    """
+
+    if isinstance(input_file, str) and urlparse(input_file).scheme in ["http", "https"]:
+        html = f'<embed src="{input_file}" type="application/pdf" width="700px" height="700px"/></embed>'
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/utils/html_utils.py]: same for width and height arguments

File: src/argilla/client/feedback/utils/html_utils.py (hunk 1 above)
Comment (sdiazlor) [reply src/argilla/client/feedback/utils/html_utils.py]: Not really, so yeah, we can allow them

Comment (davidberenstein1957) [reply src/argilla/client/feedback/utils/html_utils.py]: can you do some additional testing and add these as function arguments? Potentially too for the other methods but feel free to assess if it is relevant there too 👍

File: src/argilla/client/feedback/utils/html_utils.py [hunk 4]
```diff
@@ -28,121 +30,243 @@
...
 
     Examples:
         >>> from argilla.client.feedback.utils import image_to_html
-        >>> html = image_to_html("my_image.png")
+        >>> html = image_to_html("my_image.png", width="300px", height="300px")
+    """
+    return media_to_html("image", file_source, file_type, width, height)
+
+
+def pdf_to_html(file_source: Union[str, bytes], width: Optional[str] = "700px", height: Optional[str] = "700px") -> str:
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/utils/html_utils.py]: should we set this to 100% by default? @sdiazlor

**Review by davidberenstein1957 — APPROVED** [review]

**Comment by davidberenstein1957** [comment]
Only this last comment, otherwise you can merge :)

**Pull request merged**